### ISSUE TITLE
Loss Calculator

### ISSUE THREAD

**Issue by psjanani** [neulab/xnmt#216]
Specifying the ```loss_calculator``` within the model seems like a good idea but it is binding the model to a specific training process as we get the entire model specifications from the pre-trained model. This prevents us from choosing to fine-tune a model with a training process other than it was initially trained with.
Should this be made more flexible? Or am I missing something?

**Comment by msperber** [comment]
I also think changing parts of a pretrained model would be nice to have! One suggestions would be to make the config file for finetuning look something like:
exp1:
  train:
    model: !PretrainedModel
      location: /path/to/pretrained.mod
      overwrite:
      - name: translator
        value: !TranslatorReinforceLoss {}
      - name: encoder.modules.0.dropout
        value: 0.1

So we would need to load the serialized YAML file (without initializing components), adjust the parts that are overwritten, and then initialize everything. This will get more complicated once we change components that have DyNet parameters, but I guess we could ignore that case for the first attempt..

**Comment by neubig** [comment]
I think these are two different things. I totally agree with @psjanani 's decoupling of the training strategy and the model. @msperber 's overwriting part of the model also sounds like it might be useful, but it'd be more work. I say we do the former immediately, and think about how to do the latter for the future (when it becomes necessary).

**Comment by philip30** [comment]
If there is a way to overwrite or initialize the loaded model, we won't have to change the design as much. I actually have the same problem but in other things.

**Comment by neubig** [comment]
Sure, I think it would be useful to be able to do this. But it's also important for the training strategy and model to not be conflated.

**Comment by psjanani** [comment]
Awesome, I can work on decoupling training strategy and model if it is alright.
And overwriting the model would be really useful. I have been using not so pretty hacks to achieve that till now.
